dagfactory/dagfactory.py

```diff
@@ -36,6 +37,25 @@ def __init__(self, config_filepath: Optional[str] = None, config: Optional[dict]
         if config:
             self.config: Dict[str, Any] = config
 
+    @staticmethod
+    def _global_default_args():
+        default_args_yml = Path(settings.DAGS_FOLDER) / "defaults.yml"
+
+        if default_args_yml.exists():
+            with open(default_args_yml, "r") as file:
+                return yaml.safe_load(file)
+
+    def _merge_default_args(self, global_default_args, local_default_args):
+        result = global_default_args.copy()
+
+        for key, value in local_default_args.items():
+            if isinstance(value, dict) and key in result and isinstance(result[key], dict):
+                result[key] = self._merge_default_args(result[key], value)
+            else:
+                result[key] = value
```

Comment:
Did you have any particular use-case in mind to add this logic?

I feel we should not attempt to merge the values of the "duplicate" keys in different default configurations. This could lead to more confusion.

We can override keys at the top level, non-recursively, following our order of precedence.

Reply:
I thought we have nested default_args, but we don't have nested default_args. I have removed this method and just updating now